<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Prepare a BulkSignalR data set — prepareDataset • BulkSignalR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Prepare a BulkSignalR data set — prepareDataset" />
<meta property="og:description" content="Take a matrix or data frame containing read counts along with
  sample types and combine them into a list for subsequent use with
  BulkSignalR. Expression proteomics data could be used as well." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BulkSignalR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/UserGuide.html">User's Guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/zhefrench/BulkSignalR/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prepare a BulkSignalR data set</h1>
    <small class="dont-index">Source: <a href='https://github.com/zhefrench/BulkSignalR/blob/master/R/dataPrepare.R'><code>R/dataPrepare.R</code></a></small>
    <div class="hidden name"><code>prepareDataset.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Take a matrix or data frame containing read counts along with
  sample types and combine them into a list for subsequent use with
  BulkSignalR. Expression proteomics data could be used as well.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>prepareDataset</span><span class='op'>(</span>
  <span class='va'>counts</span>,
  <span class='va'>types</span>,
  normalize <span class='op'>=</span> <span class='cn'>TRUE</span>,
  symbol.col <span class='op'>=</span> <span class='cn'>NULL</span>,
  min.count <span class='op'>=</span> <span class='fl'>10</span>,
  prop <span class='op'>=</span> <span class='fl'>0.1</span>,
  method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"UQ"</span>, <span class='st'>"TC"</span><span class='op'>)</span>,
  log.transformed <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>counts</th>
      <td><p>A table or matrix of read counts.</p></td>
    </tr>
    <tr>
      <th>types</th>
      <td><p>A vector of characters defining sample types.</p></td>
    </tr>
    <tr>
      <th>normalize</th>
      <td><p>A logical indicating whether <code>counts</code> should be
normalized according to <code>method</code> or it was normalized beforehand.</p></td>
    </tr>
    <tr>
      <th>symbol.col</th>
      <td><p>The index of the column containing the gene symbols in case
those are not the rownames of <code>counts</code> already.</p></td>
    </tr>
    <tr>
      <th>min.count</th>
      <td><p>The minimum read count of a gene to be considered expressed
in a sample.</p></td>
    </tr>
    <tr>
      <th>prop</th>
      <td><p>The minimum proportion of samples where a must be expressed
to keep that gene.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>The normalization method ('UQ' for upper quartile or 'TC'
for total count).</p></td>
    </tr>
    <tr>
      <th>log.transformed</th>
      <td><p>A logical indicating whether expression data were
already log-transformed, e.g., some microarray data.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list containing the read counts and the sample types for further
  use in BulkSignalR.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The <code>counts</code> matrix or table should be provided with expression
  levels of protein coding genes in each samples (column) and
  <code>rownames(counts)</code> set to the gene symbols. For commodity, it is also
  possible to provide the expression matrix or table <code>counts</code> with the
  gene symbols stored in one of the columns. This column must be specified
  with <code>symbol.col</code>. In such a case, <code>prepareDataset</code> will extract
  this column and use it to set the row names. Because row names must be
  unique, <code>prepareDataset</code> will eliminate rows with duplicated gene
  symbols by keeping the rows with maximum average expression. Gene symbol
  duplication typically occurs in protein coding genes after genome alignment
  due to errors in genome feature annotation files (GTF/GFF), where a handful
  of deprecated gene annotations may remain along with the current ones, or
  some genes are not given their fully specific symbols. If your read count
  extraction pipeline does not take care of this phenomenon, the maximum mean
  expression selection strategy implemented here should solve this difficulty
  for the sake of inferring ligand-receptor interactions.</p>
<p>If <code>normalize</code> is <code>TRUE</code> then normalization is performed
  according to <code>method</code>. If those two simple methods are not satisfying,
  then it is possible to provide a pre-normalized matrix, e.g., using edgeR
  TMM algorithm.</p>
<p>In case proteomic data are provided, <code>min.count</code> must be understood as
  its equivalent with respect to those data (spectral counts, iBAQ, etc.).</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/print.html'>print</a></span><span class='op'>(</span><span class='st'>'prepareDataset'</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "prepareDataset"</span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>sdc</span>,package<span class='op'>=</span><span class='st'>'BulkSignalR'</span><span class='op'>)</span></span>
<span class='r-in'><span class='va'>sample.types</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='st'>'tumor'</span>,<span class='fu'><a href='https://rdrr.io/r/base/nrow.html'>ncol</a></span><span class='op'>(</span><span class='va'>sdc</span><span class='op'>)</span><span class='op'>)</span></span>
<span class='r-in'><span class='va'>sample.types</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/grep.html'>grep</a></span><span class='op'>(</span><span class='st'>'^N'</span>,<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='va'>sdc</span><span class='op'>)</span>,perl<span class='op'>=</span><span class='cn'>TRUE</span><span class='op'>)</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='st'>'normal'</span></span>
<span class='r-in'><span class='va'>ds</span> <span class='op'>&lt;-</span> <span class='fu'>prepareDataset</span><span class='op'>(</span><span class='va'>sdc</span>,<span class='va'>sample.types</span><span class='op'>)</span></span>
<span class='r-in'></span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Jacques Colinge, Jean-Philippe Villemin.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.9000.9000.9000.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


