<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="BulkSignalR">
<title>`BulkSignalR` :  Inference of ligand-receptor interactions from bulk-rnaseq. • BulkSignalR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="`BulkSignalR` : &lt;/br&gt; Inference of ligand-receptor interactions from bulk-rnaseq.">
<meta property="og:description" content="BulkSignalR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">BulkSignalR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.9</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/BulkSignalR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jcolinge/BulkSignalR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="BulkSignalR_files/kePrint-0.0.1/kePrint.js"></script><link href="BulkSignalR_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>`BulkSignalR` :  Inference of ligand-receptor interactions from bulk-rnaseq.</h1>
                        <h4 data-toc-skip class="author">Jean-Philippe Villemin</h4>
            <address class="author_afil">
      Institut de Recherche en Cancérologie de Montpellier, Inserm, Montpellier, France<br><a class="author_email" href="mailto:#"></a><a href="mailto:jean-philippe.villemin@inserm.fr">jean-philippe.villemin@inserm.fr</a>
      </address>
                              <h4 data-toc-skip class="author">Jacques Colinge</h4>
            <address class="author_afil">
      Institut de Recherche en Cancérologie de Montpellier, Inserm, Montpellier, France<br><a class="author_email" href="mailto:#"></a><a href="mailto:jacques.colinge@inserm.fr">jacques.colinge@inserm.fr</a>
      </address>
                  
            <h4 data-toc-skip class="date">07/22/2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jcolinge/BulkSignalR/blob/HEAD/vignettes/BulkSignalR.Rmd" class="external-link"><code>vignettes/BulkSignalR.Rmd</code></a></small>
      <div class="d-none name"><code>BulkSignalR.Rmd</code></div>
    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>
      </p>
<p>BulkSignalR is used to infer ligand-receptor (LR) interactions from bulk (transcriptomic or proteomic) data. BulkSignalR bases its inferences on the LRdb database included in our other package, SingleCellSignalR available from Bioconductor. It relies on a statistical model that is specific to bulk data sets. Different visualization and data summary functions are proposed to help navigating prediction results.<br><br> BulkSignalR package version: 0.0.9</p>
    </div>
    
<pre class="{=html}"><code>&lt;style type="text/css"&gt;
.smaller {
  font-size: 10px
}
&lt;/style&gt;</code></pre>
<div class="section level2">
<h2 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#devtools::install_github("BulkSignalR")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/devtools/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"/home/jp/Documents/workingDir/new/BulkSignalR"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jcolinge/BulkSignalR" class="external-link">BulkSignalR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/" class="external-link">kableExtra</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="section level3">
<h3 id="what-is-it-for">What is it for ?<a class="anchor" aria-label="anchor" href="#what-is-it-for"></a>
</h3>
<p><code>BulkSignalR</code> is is a tool that enables the detection of Ligand-Receptor (LR) interactions.</p>
<p>Originally developed for bulk RNA-SEQ, its application can be extended to spatial transcriptomics as 10X’s Visium technology,</p>
</div>
<div class="section level3">
<h3 id="starting-point">Starting point<a class="anchor" aria-label="anchor" href="#starting-point"></a>
</h3>
<p>There are a variety of steps upstream of <code>BulkSignalR</code> that result in the generation of counts or normalized counts (RPKM, CPM, TPM...) for each sample, which can be seamlessly used in our workflow.</p>
<p>The minimal unit of data required to run <code>BulksignalR</code> is : <br><br>
- A matrix with the expression where rows = genes and columns = samples. The first column of the table must contain gene names as HUGO symbols.</p>
</div>
<div class="section level3">
<h3 id="how-it-works">How it works ?<a class="anchor" aria-label="anchor" href="#how-it-works"></a>
</h3>
<p><code>BulkSignalR</code> uses (TODO).</p>
<p>As described on figure below, a few steps are required in order to process a <code>BulkSignalR</code> analysis.</p>
<p>Three S4 objects will be sequentially constructed as follows :<br>
* <strong>BSR-DataModel</strong>, annotated as <code>bsrdm</code><br>
* <strong>BSR-Inference</strong>, annotated as <code>bsrinf</code><br>
* <strong>BSR-Signature</strong>, annotated as <code>bsrsig</code></p>
<div class="figure">
<img src="../reference/figures/workflow.png" alt="id0"><p class="caption">id0</p>
</div>
<p><strong>BSR-DataModel</strong> will integrates the raw matrix and stores the learned parameters of the LR correlation distribution.</p>
<p><strong>BSR-Inference</strong> will provide the list of LR interactions annotated with their potential underlying pathways. As a first result, you can easily export this table using <code>LRinter</code> function.</p>
<p><strong>BSR-Signature</strong> will compute gene signature scores for each samples, by focusing either on LR interactions or on pathways. Finally, user can visualize these scoring via heatmaps to highlight some patterns in the data.</p>
<p>Ligand-Receptor interaction might contain redundant information that can be filtered out using several reduction strategies (<code>reduceToPathway</code>,<code>reduceToBestPathway</code>,...) detailed in the following sections.</p>
<p>Furthermore, we provide several handy functions to explore the data via different plots (heatmap, alluvial plot, chordiagram or networks).</p>
<p><code>BulkSignalR</code> package functions have many arguments parameters that can be changed by the user to fit its needs (see Reference Manual for more details).</p>
</div>
<div class="section level3">
<h3 id="parallel-mode-settings">Parallel mode settings<a class="anchor" aria-label="anchor" href="#parallel-mode-settings"></a>
</h3>
<p>User can simply increases the speed of an analysis by using several cores in parallel. For this purpose, one can set the number of cores he wants to use before starting <code>BulkSignalR</code>.</p>
<p>In parallel mode, an <code>iseed</code> can be provided with <code>clusterSetRNGStream</code> in order to reproduce the exact same analysis due to sampling In standard mode, <code><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed()</a></code> is sufficient.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RevolutionAnalytics/doparallel" class="external-link">doParallel</a></span><span class="op">)</span></span>
<span><span class="va">n.proc</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span><span class="va">cl</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/parallel/makeCluster.html" class="external-link">makeCluster</a></span><span class="op">(</span><span class="va">n.proc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/parallel/RngStream.html" class="external-link">clusterSetRNGStream</a></span><span class="op">(</span><span class="va">cl</span>, iseed <span class="op">=</span> <span class="fl">1983</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/doParallel/man/registerDoParallel.html" class="external-link">registerDoParallel</a></span><span class="op">(</span><span class="va">cl</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># To add at the end of your script</span></span>
<span><span class="co"># stopCluster(cl)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="first-example">First example<a class="anchor" aria-label="anchor" href="#first-example"></a>
</h2>
<div class="section level3">
<h3 id="loading-the-data">Loading the data<a class="anchor" aria-label="anchor" href="#loading-the-data"></a>
</h3>
<p>Here, we load salivary duct carcinoma transcriptomes integrated as <code>sdc</code> in <code>BulkSignalR</code>.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sdc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sdc</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          SDC16 SDC15 SDC14 SDC13 SDC12 SDC11 SDC10 SDC9 SDC8 SDC7 SDC6 SDC5</span></span>
<span><span class="co">## TSPAN6    1141  2491  1649  3963  2697  3848  3115  495 1777 1997 1682 1380</span></span>
<span><span class="co">## TNMD         3     0     2    48     0     0    10    0  195    0    4    0</span></span>
<span><span class="co">## DPM1      1149  2105  1579  2597  2569  2562  1690  964  980 3323 3409  743</span></span>
<span><span class="co">## SCYL3     1814   768  3513  1402  1007  1243  2501 3080  939 1190 1489 1097</span></span>
<span><span class="co">## C1orf112   438  1016   507  1149   598  1017   464  531  388 1088 1719  625</span></span>
<span><span class="co">## FGR        381   922   307   256   249   902   304  355  410 1054  265  121</span></span>
<span><span class="co">##          SDC4 SDC3 SDC2 SDC1 SDC17 SDC19 SDC20  N20 SDC21 SDC22  N22 SDC23</span></span>
<span><span class="co">## TSPAN6   1392 2490 1841 1883  1915  4093  3127 8035  4738  2994 9782  3866</span></span>
<span><span class="co">## TNMD        0    0    0    1     5     0    18   21     6     2   27     0</span></span>
<span><span class="co">## DPM1     1147 1705 1972 3048  2405  2943  2710 1688  2654  2772 2018  1610</span></span>
<span><span class="co">## SCYL3     507 1545 2450 2006   910  1842  2947 1058  2636  4199 1173  2980</span></span>
<span><span class="co">## C1orf112  318 1338 3065  723   509  1320   767  179   718  1498  147   629</span></span>
<span><span class="co">## FGR       408  813  787 1152  1049  4205   722  149   584   701  164   842</span></span>
<span><span class="co">##          SDC24 SDC25</span></span>
<span><span class="co">## TSPAN6    3014  1765</span></span>
<span><span class="co">## TNMD        14     3</span></span>
<span><span class="co">## DPM1      2875  2357</span></span>
<span><span class="co">## SCYL3     2079   903</span></span>
<span><span class="co">## C1orf112  1008   521</span></span>
<span><span class="co">## FGR        676   654</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="building-a-bsr-datamodel">Building a BSR-DataModel<a class="anchor" aria-label="anchor" href="#building-a-bsr-datamodel"></a>
</h3>
<p><code>prepareDataset</code> will create the first object with information relative to your experiment.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bsrdm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepareDataset.html">prepareDataset</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">sdc</span><span class="op">)</span></span>
<span><span class="co"># print object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">bsrdm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Formal class 'BSRDataModel' [package "BulkSignalR"] with 6 slots</span></span>
<span><span class="co">##   ..@ initial.organism : chr "hsapiens"</span></span>
<span><span class="co">##   ..@ initial.orthologs: list()</span></span>
<span><span class="co">##   ..@ ncounts          : num [1:16822, 1:26] 1286 3.38 1295.01 2044.52 493.66 ...</span></span>
<span><span class="co">##   .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">##   .. .. ..$ : chr [1:16822] "TSPAN6" "TNMD" "DPM1" "SCYL3" ...</span></span>
<span><span class="co">##   .. .. ..$ : chr [1:26] "SDC16" "SDC15" "SDC14" "SDC13" ...</span></span>
<span><span class="co">##   ..@ log.transformed  : logi FALSE</span></span>
<span><span class="co">##   ..@ normalization    : chr "UQ"</span></span>
<span><span class="co">##   ..@ param            : list()</span></span></code></pre>
<p><code>learnParameters</code> will be updated with parameters of LR correlation distribution. You can specify the directory and the file name where you want to plot this distribution.</p>
<p><code>null.model</code> is set automatically according to the shape of this distribution.</p>
<p>However <code>BulkSignalR</code> authorize three modes for this parameter :<br>
* <em>normal</em> : for most cases, gaussian distribution.<br>
* <em>mixedNormal</em> : when distribution looks bimodal.<br>
* <em>stable</em> : for spatial transcriptomics (ST).</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bsrdm</span> <span class="op">&lt;-</span> <span class="fu">learnParameters</span><span class="op">(</span><span class="va">bsrdm</span>, null.model <span class="op">=</span> <span class="st">"normal"</span>, quick <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>         plot.folder <span class="op">=</span> <span class="st">"../man/figures/"</span>, filename <span class="op">=</span> <span class="st">"sdc"</span>,</span>
<span>         verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># print object updated </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">bsrdm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Formal class 'BSRDataModel' [package "BulkSignalR"] with 6 slots</span></span>
<span><span class="co">##   ..@ initial.organism : chr "hsapiens"</span></span>
<span><span class="co">##   ..@ initial.orthologs: list()</span></span>
<span><span class="co">##   ..@ ncounts          : num [1:16822, 1:26] 1286 3.38 1295.01 2044.52 493.66 ...</span></span>
<span><span class="co">##   .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">##   .. .. ..$ : chr [1:16822] "TSPAN6" "TNMD" "DPM1" "SCYL3" ...</span></span>
<span><span class="co">##   .. .. ..$ : chr [1:26] "SDC16" "SDC15" "SDC14" "SDC13" ...</span></span>
<span><span class="co">##   ..@ log.transformed  : logi FALSE</span></span>
<span><span class="co">##   ..@ normalization    : chr "UQ"</span></span>
<span><span class="co">##   ..@ param            :List of 13</span></span>
<span><span class="co">##   .. ..$ n.rand.LR   : int 5</span></span>
<span><span class="co">##   .. ..$ n.rand.RT   : int 2</span></span>
<span><span class="co">##   .. ..$ plot.folder : chr "../man/figures/"</span></span>
<span><span class="co">##   .. ..$ file.name   : chr "sdc"</span></span>
<span><span class="co">##   .. ..$ with.complex: logi TRUE</span></span>
<span><span class="co">##   .. ..$ max.pw.size : num 200</span></span>
<span><span class="co">##   .. ..$ min.pw.size : num 5</span></span>
<span><span class="co">##   .. ..$ min.positive: num 4</span></span>
<span><span class="co">##   .. ..$ quick       : logi FALSE</span></span>
<span><span class="co">##   .. ..$ seed        : num 123</span></span>
<span><span class="co">##   .. ..$ min.corr.LR : num -1</span></span>
<span><span class="co">##   .. ..$ LR.0        :List of 2</span></span>
<span><span class="co">##   .. .. ..$ n    : int 12915</span></span>
<span><span class="co">##   .. .. ..$ model:List of 5</span></span>
<span><span class="co">##   .. .. .. ..$ mu     : num 0.0172</span></span>
<span><span class="co">##   .. .. .. ..$ sigma  : num 0.254</span></span>
<span><span class="co">##   .. .. .. ..$ factor : num 1</span></span>
<span><span class="co">##   .. .. .. ..$ start  : num 3.2e-05</span></span>
<span><span class="co">##   .. .. .. ..$ distrib: chr "censored_normal"</span></span>
<span><span class="co">##   .. ..$ RT.0        :List of 2</span></span>
<span><span class="co">##   .. .. ..$ n    : int 954132</span></span>
<span><span class="co">##   .. .. ..$ model:List of 5</span></span>
<span><span class="co">##   .. .. .. ..$ mu     : num 0.015</span></span>
<span><span class="co">##   .. .. .. ..$ sigma  : num 0.274</span></span>
<span><span class="co">##   .. .. .. ..$ factor : num 1</span></span>
<span><span class="co">##   .. .. .. ..$ start  : num 0.000107</span></span>
<span><span class="co">##   .. .. .. ..$ distrib: chr "censored_normal"</span></span></code></pre>
<p>The plot with distribution of LR correlation in <code>sdc</code> is displayed below.</p>
<div class="figure">
<img src="../reference/figures/sdc_LR-null.png" alt="id1"><p class="caption">id1</p>
</div>
</div>
<div class="section level3">
<h3 id="building-a-bsr-inference">Building a BSR-Inference<a class="anchor" aria-label="anchor" href="#building-a-bsr-inference"></a>
</h3>
<p>From the previous object <code>bsrdm</code> , you will create <code>bsrinf</code> by calling <code>initialInference</code> . <code>bsrinf</code> contains the inferred LR interactions with their associated pathways and corrected p-values.</p>
<p>From here, you can already access LR interactions using : <code>LRinter(bsrinf)</code>.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bsrinf</span> <span class="op">&lt;-</span> <span class="fu">initialInference</span><span class="op">(</span><span class="va">bsrdm</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LRinter.dataframe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">LRinter</span><span class="op">(</span><span class="va">bsrinf</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">LRinter.dataframe</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">LRinter.dataframe</span><span class="op">$</span><span class="va">qval</span> <span class="op">&lt;=</span> <span class="fl">0.001</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L"</span>,    <span class="st">"R"</span> ,<span class="st">"LR.corr"</span>, <span class="st">"pw.id"</span>,    <span class="st">"pw.name"</span>,  <span class="st">"pval"</span>  ,   <span class="st">"qval"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>,row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_classic.html" class="external-link">kable_material</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"striped"</span>, <span class="st">"hover"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table  lightable-material lightable-striped lightable-hover" style='font-family: "Source Sans Pro", helvetica, sans-serif; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
L
</th>
<th style="text-align:left;">
R
</th>
<th style="text-align:right;">
LR.corr
</th>
<th style="text-align:left;">
pw.id
</th>
<th style="text-align:left;">
pw.name
</th>
<th style="text-align:right;">
pval
</th>
<th style="text-align:right;">
qval
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
A2M
</td>
<td style="text-align:left;">
LRP1
</td>
<td style="text-align:right;">
0.5391453
</td>
<td style="text-align:left;">
R-HSA-196854
</td>
<td style="text-align:left;">
Metabolism of vitamins and cofactors
</td>
<td style="text-align:right;">
0.0200711
</td>
<td style="text-align:right;">
0.0260773
</td>
</tr>
<tr>
<td style="text-align:left;">
A2M
</td>
<td style="text-align:left;">
LRP1
</td>
<td style="text-align:right;">
0.5391453
</td>
<td style="text-align:left;">
R-HSA-2173782
</td>
<td style="text-align:left;">
Binding and Uptake of Ligands by Scavenger Receptors
</td>
<td style="text-align:right;">
0.0181048
</td>
<td style="text-align:right;">
0.0239389
</td>
</tr>
<tr>
<td style="text-align:left;">
A2M
</td>
<td style="text-align:left;">
LRP1
</td>
<td style="text-align:right;">
0.5391453
</td>
<td style="text-align:left;">
R-HSA-2187338
</td>
<td style="text-align:left;">
Visual phototransduction
</td>
<td style="text-align:right;">
0.0174728
</td>
<td style="text-align:right;">
0.0232013
</td>
</tr>
<tr>
<td style="text-align:left;">
A2M
</td>
<td style="text-align:left;">
LRP1
</td>
<td style="text-align:right;">
0.5391453
</td>
<td style="text-align:left;">
R-HSA-6806667
</td>
<td style="text-align:left;">
Metabolism of fat-soluble vitamins
</td>
<td style="text-align:right;">
0.0112186
</td>
<td style="text-align:right;">
0.0161057
</td>
</tr>
<tr>
<td style="text-align:left;">
A2M
</td>
<td style="text-align:left;">
LRP1
</td>
<td style="text-align:right;">
0.5391453
</td>
<td style="text-align:left;">
R-HSA-975634
</td>
<td style="text-align:left;">
Retinoid metabolism and transport
</td>
<td style="text-align:right;">
0.0112186
</td>
<td style="text-align:right;">
0.0161057
</td>
</tr>
<tr>
<td style="text-align:left;">
ADAM10
</td>
<td style="text-align:left;">
AXL
</td>
<td style="text-align:right;">
0.4441026
</td>
<td style="text-align:left;">
R-HSA-194138
</td>
<td style="text-align:left;">
Signaling by VEGF
</td>
<td style="text-align:right;">
0.0029121
</td>
<td style="text-align:right;">
0.0056264
</td>
</tr>
</tbody>
</table>
<p><br> You can finally filter out the less significant LR interactions and order them by best q-value before saving them to a file.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">LRinter.dataframe</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">LRinter.dataframe</span><span class="op">$</span><span class="va">qval</span> <span class="op">&lt;=</span> <span class="fl">0.001</span><span class="op">)</span>,<span class="op">]</span>,</span>
<span>            <span class="st">"../man/figures/LR.tsv"</span>,</span>
<span>            row.names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>            quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="reduction-strategies">Reduction strategies<a class="anchor" aria-label="anchor" href="#reduction-strategies"></a>
</h3>
<p>Raw results from <code>LRinter(bsrinf)</code> can suffer from redundancy. LR interactions can appear several times in several pathways.</p>
<p>To resolve this pitfall, we propose several ways to reduce information in order to increase readability of the results.</p>
<p>User can simplify a <code>bsrinf</code> object on demand, to focus etheir on the pathways or on LR interactions.</p>
<div class="section level4">
<h4 id="reducing-a-bsrinference-to-pathway">Reducing a BSRInference to pathway<a class="anchor" aria-label="anchor" href="#reducing-a-bsrinference-to-pathway"></a>
</h4>
<p>With <code>reduceToPathway</code> ,all the ligands and receptors forming pairs related to a certain pathway, are pulled together to only report one one row per pathway. For a given pathway, the reported P-values and target genes are those of the best LR pair that appeared in this pathway.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>     <span class="va">bsrinf.redP</span> <span class="op">&lt;-</span> <span class="fu">reduceToPathway</span><span class="op">(</span><span class="va">bsrinf</span><span class="op">)</span>  </span>
<span>     <span class="co"># write.table(as.data.frame(LRinter(bsrinf.redP)), </span></span>
<span>     <span class="co"># "../man/figures/sdc_LR.reduceToPathway.tsv", </span></span>
<span>     <span class="co"># row.names = FALSE, sep="\t", quote=FALSE)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="reducing-a-bsrinference-to-best-pathway">Reducing a BSRInference to best pathway<a class="anchor" aria-label="anchor" href="#reducing-a-bsrinference-to-best-pathway"></a>
</h4>
<p>With <code>ReduceToBestPathway</code>, a BSR-Inference object is reduced to only report one pathway per LR pair. The pathway with the smallest P-value is selected for the pair. Pathways can be on several rows associated with different LR interactions.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>     <span class="va">bsrinf.redBP</span>    <span class="op">&lt;-</span> <span class="fu">reduceToBestPathway</span><span class="op">(</span><span class="va">bsrinf</span><span class="op">)</span></span>
<span>     </span>
<span>     <span class="co"># write.table(as.data.frame(LRinter(bsrinf.red)),</span></span>
<span>     <span class="co"># "../man/figures/sdc_LR.reduceToBestPathway.tsv", </span></span>
<span>     <span class="co"># row.names = FALSE, sep="\t", quote=FALSE)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="double-reduction">Double Reduction<a class="anchor" aria-label="anchor" href="#double-reduction"></a>
</h4>
<p>User can apply <code>reduceToPathway(bsrinf)</code> and <code>reduceToBestPathway(bsrinf)</code> reductions sequentially to maximize reduction effect.</p>
<p>If the exact same pool of LR given by <code>reduceToPathway(bsrinf)</code> is associated with several pathways, the pathway with the best p-value for this pool will be kept by <code>reduceToBestPathway(bsrinf)</code>.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>     <span class="va">bsrinf.redP</span>  <span class="op">&lt;-</span> <span class="fu">reduceToPathway</span><span class="op">(</span><span class="va">bsrinf</span><span class="op">)</span>  </span>
<span>     <span class="va">bsrinf.redPBP</span> <span class="op">&lt;-</span> <span class="fu">reduceToBestPathway</span><span class="op">(</span><span class="va">bsrinf.redP</span><span class="op">)</span> </span>
<span>     </span>
<span>     <span class="co"># write.table(as.data.frame(LRinter(bsrinf.redPBP)),</span></span>
<span>     <span class="co"># "../man/figures/sdc_LR.double.reduction.tsv", </span></span>
<span>     <span class="co"># row.names = FALSE, sep="\t", quote=FALSE)</span></span></code></pre></div>
<p>These reductions will be useful in order to compute gene signature and visualize our data. This will be explained below.</p>
</div>
</div>
<div class="section level3">
<h3 id="building-a-bsr-signature">Building a BSR-Signature<a class="anchor" aria-label="anchor" href="#building-a-bsr-signature"></a>
</h3>
<p>Finally we applied <code>getLRGeneSignatures</code> with the different BSR-Inference object previously reduced. Thus, we will create several <code>BSR-Signature object</code> <code>bsrsig.redPP</code>.</p>
<p><code>scoreLRGeneSignatures</code> is applied conjointly on BSR-Inference &amp; BSR-Signature objects to score signatures at the pathway or LR interaction levels.</p>
<p>Then, we provide a <code>simpleHeatmap</code> function to visualize gene score signatures inside a heatmap as shown below.</p>
<div class="section level4">
<h4 id="scoring-by-ligand-receptor">Scoring by ligand-receptor<a class="anchor" aria-label="anchor" href="#scoring-by-ligand-receptor"></a>
</h4>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>     <span class="va">bsrsig.redBP</span> <span class="op">&lt;-</span> <span class="fu">getLRGeneSignatures</span><span class="op">(</span><span class="va">bsrinf.redBP</span>, qval.thres<span class="op">=</span><span class="fl">0.001</span><span class="op">)</span></span>
<span></span>
<span>     <span class="va">scoresLR</span> <span class="op">&lt;-</span> <span class="fu">scoreLRGeneSignatures</span><span class="op">(</span><span class="va">bsrdm</span>,<span class="va">bsrsig.redBP</span>,name.by.pathway<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span>     </span>
<span>     <span class="fu"><a href="../reference/simpleHeatmap.html">simpleHeatmap</a></span><span class="op">(</span><span class="va">scoresLR</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>,<span class="op">]</span>, </span>
<span>                   path <span class="op">=</span> <span class="st">"../man/figures/"</span>,</span>
<span>                   filename <span class="op">=</span> <span class="st">"sdc_scoresLR"</span>,</span>
<span>                   column.names <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                   height <span class="op">=</span> <span class="fl">5</span>, width <span class="op">=</span> <span class="fl">9</span>,</span>
<span>                   pointsize <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                   hcl_palette <span class="op">=</span> <span class="st">"Cividis"</span>                   </span>
<span>                   <span class="op">)</span></span>
<span>     </span>
<span>   </span>
<span></span>
<span>    <span class="co"># You can easily export the scoring matrix for LR as follows</span></span>
<span>    <span class="co"># write.table(scoresLR, </span></span>
<span>    <span class="co"># "../man/figures/sdc_scoreLR.tsv",</span></span>
<span>    <span class="co"># col.names = NA,sep = "\t",</span></span>
<span>    <span class="co"># quote = FALSE)</span></span></code></pre></div>
<div class="figure">
<img src="../reference/figures/sdc_scoresLR.svg" alt="id3"><p class="caption">id3</p>
</div>
</div>
<div class="section level4">
<h4 id="scoring-by-pathway">Scoring by pathway<a class="anchor" aria-label="anchor" href="#scoring-by-pathway"></a>
</h4>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>     <span class="va">bsrsig.redPBP</span> <span class="op">&lt;-</span> <span class="fu">getLRGeneSignatures</span><span class="op">(</span><span class="va">bsrinf.redPBP</span>,qval.thres<span class="op">=</span><span class="fl">0.001</span><span class="op">)</span></span>
<span>     </span>
<span>     <span class="va">scoresPathway</span> <span class="op">&lt;-</span> <span class="fu">scoreLRGeneSignatures</span><span class="op">(</span><span class="va">bsrdm</span>,<span class="va">bsrsig.redPBP</span>,name.by.pathway<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span>     </span>
<span>     <span class="fu"><a href="../reference/simpleHeatmap.html">simpleHeatmap</a></span><span class="op">(</span><span class="va">scoresPathway</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span>, </span>
<span>                   path <span class="op">=</span> <span class="st">"../man/figures/"</span>,</span>
<span>                   filename <span class="op">=</span> <span class="st">"sdc_scoresPathway"</span>,</span>
<span>                   column.names <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                   width <span class="op">=</span> <span class="fl">9</span>, </span>
<span>                   height <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                   pointsize <span class="op">=</span> <span class="fl">12</span>,</span>
<span>                   hcl_palette <span class="op">=</span> <span class="st">"Blue-Red 2"</span></span>
<span>                   <span class="op">)</span></span>
<span></span>
<span>     <span class="co"># Dual Heatmap for comparing pre-existing signature</span></span>
<span>     <span class="co"># data(immune.signatures, package="BulkSignalR")</span></span>
<span>     <span class="co"># imm.scores &lt;- scoreSignatures(bsrdm, immune.signatures)</span></span>
<span>     <span class="co"># dualHeatmap(scoresPathway, imm.scores, width=6, height=9,</span></span>
<span>     <span class="co"># file="SDC-LR-dualheatmap.pdf",</span></span>
<span>     <span class="co"># pointsize=4, vert.p=0.85)</span></span>
<span>     </span>
<span>     <span class="co"># You can easily export the scoring matrix for pathways as follows</span></span>
<span>     <span class="co"># write.table(scoresPathway,</span></span>
<span>     <span class="co"># "../man/figures/sdc_scorepathways.tsv",</span></span>
<span>     <span class="co"># col.names = NA,sep = "\t",</span></span>
<span>     <span class="co"># quote = FALSE)</span></span></code></pre></div>
<div class="figure">
<img src="../reference/figures/sdc_scoresPathway.svg" alt="id4"><p class="caption">id4</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="other-visualisation-utilities">Other Visualisation utilities<a class="anchor" aria-label="anchor" href="#other-visualisation-utilities"></a>
</h3>
<p>In the next section, we show how to visualize the data in different ways.</p>
<div class="section level4">
<h4 id="heatmap-expression-ligand-receptor-target">Heatmap Expression Ligand-Receptor-Target<a class="anchor" aria-label="anchor" href="#heatmap-expression-ligand-receptor-target"></a>
</h4>
<p>Once you have compute gene signatures score. One may wish to look at the expression of the genes involved in that signature.</p>
<p>For instance, here we display three heatmaps corresponding to the z-scored expression of Ligands (Pink), Receptors (Green) and Targets (Blue).<br>
On top, user can see the original scoring computed before by <code>scoreLRGeneSignatures</code> for the given pathway.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathway1</span> <span class="op">&lt;-</span> <span class="st">"Elastic fibre formation"</span></span>
<span><span class="fu"><a href="../reference/signatureHeatmaps.html">signatureHeatmaps</a></span><span class="op">(</span></span>
<span>        pathway <span class="op">=</span> <span class="va">pathway1</span>,</span>
<span>        bsrdm <span class="op">=</span> <span class="va">bsrdm</span>,</span>
<span>        bsrsig <span class="op">=</span> <span class="va">bsrsig.redPBP</span>,</span>
<span>        path <span class="op">=</span> <span class="st">"../man/figures/"</span>,</span>
<span>        filename <span class="op">=</span> <span class="st">"sdc_signatureheatmap"</span>,</span>
<span>        width  <span class="op">=</span> <span class="fl">15</span>,</span>
<span>        height <span class="op">=</span> <span class="fl">10</span> ,</span>
<span>        show_column_names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="../reference/figures/sdc_signatureheatmap.svg" alt="id5"><p class="caption">id5</p>
</div>
</div>
<div class="section level4">
<h4 id="alluvialplot">AlluvialPlot<a class="anchor" aria-label="anchor" href="#alluvialplot"></a>
</h4>
<p><code>alluvial.plot</code> is a function that let the user observed the different links between ligands, receptors and pathways stored inside <code>bsrinf</code> object.</p>
<p>First you need to specify if you want to filter by Ligand, Receptor or Pathway. Then, you have to pass a keyword (gene name or pathway) to keep only the links related to your choice.</p>
<p>You also can apply a filter on Q-value to limit the number of links and improve rendering.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/alluvial.plot.html">alluvial.plot</a></span><span class="op">(</span><span class="va">bsrinf</span>,</span>
<span>              keywords <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"COL4A1"</span><span class="op">)</span>,</span>
<span>              type <span class="op">=</span> <span class="st">"L"</span>,</span>
<span>              qval <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>              path <span class="op">=</span> <span class="st">"../man/figures/"</span>,</span>
<span>              filename <span class="op">=</span> <span class="st">"sdc_alluvial"</span>, </span>
<span>              width  <span class="op">=</span> <span class="fl">16</span>, </span>
<span>              height <span class="op">=</span> <span class="fl">12</span></span>
<span>              <span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="../reference/figures/sdc_alluvial.svg" alt="id6"><p class="caption">id6</p>
</div>
</div>
<div class="section level4">
<h4 id="bubbleplot">BubblePlot<a class="anchor" aria-label="anchor" href="#bubbleplot"></a>
</h4>
<p><code>bubblePlot.pathways.LR</code> is a handy way to visualize strength of several LR interactions in relation with their associated pathways.</p>
<p>You have to give a vector of pathways.</p>
<p>Do not add too many pathways. otherwise the graph will be unreadable.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pathway1</span> <span class="op">&lt;-</span> <span class="st">"PD-1 signaling"</span></span>
<span><span class="va">pathway2</span> <span class="op">&lt;-</span> <span class="st">"Interferon gamma signaling"</span></span>
<span><span class="va">pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">pathway1</span>,<span class="va">pathway2</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/bubblePlot.pathways.LR.html">bubblePlot.pathways.LR</a></span><span class="op">(</span><span class="va">bsrinf</span>,</span>
<span>    pathways <span class="op">=</span> <span class="va">pathways</span>, </span>
<span>    threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    path <span class="op">=</span> <span class="st">"../man/figures/"</span>,</span>
<span>    color <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>    filename  <span class="op">=</span> <span class="st">"sdc_bubble"</span>, </span>
<span>    width  <span class="op">=</span> <span class="fl">16</span>, </span>
<span>    height <span class="op">=</span> <span class="fl">7</span>,</span>
<span>    pointsize <span class="op">=</span> <span class="fl">8</span></span>
<span>    <span class="co">#filter.L = c("ADAM12")</span></span>
<span>    <span class="co">#filter.R = c("ITGA3")</span></span>
<span>    <span class="op">)</span> </span></code></pre></div>
<div class="figure">
<img src="../reference/figures/sdc_bubble.svg" alt="id7"><p class="caption">id7</p>
</div>
</div>
<div class="section level4">
<h4 id="chordiagram">Chordiagram<a class="anchor" aria-label="anchor" href="#chordiagram"></a>
</h4>
<p><code>chord.diagram.LR</code> is a function that let the user observe the different LR interactions for one specific pathway.</p>
<p>LR correlations strength are drawn in yellow.<br>
Ligand are in grey, whereas Receptors are in green.<br>
You can highlight in red one specific interaction by passing values of a LR pair as follows <code>ligand="COL18A1", receptor="ITGA3"</code>.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/chord.diagram.LR.html">chord.diagram.LR</a></span> <span class="op">(</span><span class="va">bsrinf</span>,</span>
<span>                  path <span class="op">=</span> <span class="st">"../man/figures/"</span>,</span>
<span>                  filename <span class="op">=</span> <span class="st">"sdc_chord"</span>,</span>
<span>                  pw.id.filter <span class="op">=</span> <span class="st">"R-HSA-202733"</span>,</span>
<span>                  ligand <span class="op">=</span> <span class="st">"COL18A1"</span>,</span>
<span>                  receptor <span class="op">=</span> <span class="st">"ITGA3"</span>,</span>
<span>                  limit <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                  width <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                  height <span class="op">=</span> <span class="fl">4.5</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="../reference/figures/sdc_chord.svg" alt="id8"><p class="caption">id8</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="network-analysis">Network analysis<a class="anchor" aria-label="anchor" href="#network-analysis"></a>
</h2>
<p><code>BulkSignalR</code> can also map internal network linked to genes of interest through the integration of regulated KEGG and Reactome pathways together with ligands and receptors involved in inferred interactions.</p>
<p>The networks can be exported in text files and graphML objects to be further explored with Cytoscape (www.cytoscape.org), yEd (www.yworks.com), or similar software tools.</p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate a ligand-receptor network and export it in .graphML</span></span>
<span><span class="co"># for Cytoscape or similar tools</span></span>
<span><span class="va">gLR</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getLRNetwork.html">getLRNetwork</a></span><span class="op">(</span><span class="va">bsrinf.redBP</span>, qval.thres<span class="op">=</span><span class="fl">1e-8</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/write_graph.html" class="external-link">write.graph</a></span><span class="op">(</span><span class="va">gLR</span>,file<span class="op">=</span><span class="st">"SDC-LR-network.graphml"</span>,format<span class="op">=</span><span class="st">"graphml"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># As an alternative to Cytoscape, you can play with igraph package functions.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gLR</span>,</span>
<span>     edge.arrow.size<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>     vertex.label.color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>     vertex.label.family<span class="op">=</span><span class="st">"Helvetica"</span>,</span>
<span>     vertex.label.cex<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span></code></pre></div>
<p><img src="BulkSignalR_files/figure-html/network-1.png" width="700"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Community detection</span></span>
<span><span class="va">u.gLR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/as.directed.html" class="external-link">as.undirected</a></span><span class="op">(</span><span class="va">gLR</span><span class="op">)</span> <span class="co"># most algorithms work for undirected graphs only</span></span>
<span><span class="va">comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_edge_betweenness.html" class="external-link">cluster_edge_betweenness</a></span><span class="op">(</span><span class="va">u.gLR</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">comm</span>,<span class="va">u.gLR</span>,</span>
<span>     vertex.label.color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>     vertex.label.family<span class="op">=</span><span class="st">"Helvetica"</span>,</span>
<span>     vertex.label.cex<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span></code></pre></div>
<p><img src="BulkSignalR_files/figure-html/network-2.png" width="700"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Cohesive blocks</span></span>
<span><span class="va">cb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cohesive_blocks.html" class="external-link">cohesive_blocks</a></span><span class="op">(</span><span class="va">u.gLR</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">cb</span>,<span class="va">u.gLR</span>,</span>
<span>     vertex.label.color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>     vertex.label.family<span class="op">=</span><span class="st">"Helvetica"</span>,</span>
<span>     vertex.label.cex<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>     edge.color<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<p><img src="BulkSignalR_files/figure-html/network-3.png" width="700"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Generate a ligand-receptor network complemented with intra-cellular,</span></span>
<span><span class="co"># receptor downstream pathways [computations are a bit longer here]</span></span>
<span><span class="va">gLRintra</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getLRIntracellNetwork.html">getLRIntracellNetwork</a></span><span class="op">(</span><span class="va">bsrinf.redBP</span>, qval.thres<span class="op">=</span><span class="fl">1e-8</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/write_graph.html" class="external-link">write.graph</a></span><span class="op">(</span><span class="va">gLRintra</span>, file<span class="op">=</span><span class="st">"SDC-LR-intracellular-network.graphml"</span>,</span>
<span>            format<span class="op">=</span><span class="st">"graphml"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lay</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/layout_with_kk.html" class="external-link">layout_with_kk</a></span><span class="op">(</span><span class="va">gLRintra</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gLRintra</span>,</span>
<span>     layout<span class="op">=</span><span class="va">lay</span>,</span>
<span>     edge.arrow.size<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>     vertex.label.color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>     vertex.label.family<span class="op">=</span><span class="st">"Helvetica"</span>,</span>
<span>     vertex.label.cex<span class="op">=</span><span class="fl">0.1</span><span class="op">)</span></span></code></pre></div>
<p><img src="BulkSignalR_files/figure-html/network-4.png" width="700"></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Reduce complexity by focusing on strongly targeted pathways</span></span>
<span><span class="va">pairs</span> <span class="op">&lt;-</span> <span class="fu">LRinter</span><span class="op">(</span><span class="va">bsrinf.redBP</span><span class="op">)</span></span>
<span><span class="va">top</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">pairs</span><span class="op">[</span><span class="va">pairs</span><span class="op">$</span><span class="va">pval</span><span class="op">&lt;</span><span class="fl">1e-20</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pw.id"</span>,<span class="st">"pw.name"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">top</span></span>
<span><span class="va">gLRintra.res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getLRIntracellNetwork.html">getLRIntracellNetwork</a></span><span class="op">(</span><span class="va">bsrinf.redBP</span>, qval.thres<span class="op">=</span><span class="fl">0.01</span>,</span>
<span>                                      restrict.pw<span class="op">=</span><span class="va">top</span><span class="op">$</span><span class="va">pw.id</span><span class="op">)</span></span>
<span><span class="va">lay</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/layout_with_fr.html" class="external-link">layout_with_fr</a></span><span class="op">(</span><span class="va">gLRintra.res</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/write_graph.html" class="external-link">write.graph</a></span><span class="op">(</span><span class="va">gLRintra</span>, file<span class="op">=</span><span class="st">"SDC-LR-intracellular-network.reduced.graphml"</span>,</span>
<span>            format<span class="op">=</span><span class="st">"graphml"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">gLRintra.res</span>,</span>
<span>     layout<span class="op">=</span><span class="va">lay</span>,</span>
<span>     edge.arrow.size<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>     vertex.label.color<span class="op">=</span><span class="st">"black"</span>,</span>
<span>     vertex.label.family<span class="op">=</span><span class="st">"Helvetica"</span>,</span>
<span>     vertex.label.cex<span class="op">=</span><span class="fl">0.4</span><span class="op">)</span></span></code></pre></div>
<p><img src="BulkSignalR_files/figure-html/network-5.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="alternative-workflow-for-non-human-species">Alternative workflow for non human species<a class="anchor" aria-label="anchor" href="#alternative-workflow-for-non-human-species"></a>
</h2>
<p><br></p>
<p>In order to process a non human organism, user can call a few set of supplementary functions By default, <code>BulksignalR</code> works with Homo Sapiens. We developp a stragtegy using orthologous genes from the orthogene package directly integrated in BulkSignalR.</p>
<p><code>findOrthoGenes</code> create a correspondence table between human and another organism.</p>
<p><code>convertToHuman</code> will convert initial matrix to hsapiens genes nomenclature.</p>
<p>When you call <code>prepareDataset</code> , you need to pass this transformed matrix, the initial organism and the correspondence table.</p>
<p>Inference calling is done as for a normal workflow.</p>
<p>Finally, you switch back to gene names relative to the initial organism for which experiment was done via <code>resetToInitialOrganism</code> The rest of the workflow is done as usual for computing signature and visualization.</p>
<p><br></p>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">bodyMap.mouse</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ortholog.dict</span>    <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findOrthoGenes.html">findOrthoGenes</a></span> <span class="op">(</span>from_organism <span class="op">=</span> <span class="st">"mmusculus"</span>, </span>
<span>                                     from_values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">bodyMap.mouse</span><span class="op">)</span><span class="op">)</span></span>
<span> </span>
<span><span class="va">matrix.expression.human</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convertToHuman.html">convertToHuman</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">bodyMap.mouse</span>, </span>
<span>                                           dictionary <span class="op">=</span> <span class="va">ortholog.dict</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bsrdm</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../reference/prepareDataset.html">prepareDataset</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">matrix.expression.human</span>,</span>
<span>           species <span class="op">=</span> <span class="st">"mmusculus"</span>,</span>
<span>           conversion.dict <span class="op">=</span> <span class="va">ortholog.dict</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bsrdm</span> <span class="op">&lt;-</span> <span class="fu">learnParameters</span><span class="op">(</span><span class="va">bsrdm</span>, null.model<span class="op">=</span><span class="st">"normal"</span> , quick <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>           plot.folder<span class="op">=</span><span class="st">"../man/figures/"</span>,filename <span class="op">=</span> <span class="st">"bodyMap.mouse"</span>,</span>
<span>           verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span> </span>
<span><span class="va">bsrinf</span>  <span class="op">&lt;-</span> <span class="fu">initialInference</span><span class="op">(</span><span class="va">bsrdm</span><span class="op">)</span></span>
<span> </span>
<span><span class="va">bsrinf</span>  <span class="op">&lt;-</span> <span class="fu">resetToInitialOrganism</span><span class="op">(</span><span class="va">bsrinf</span>, conversion.dict<span class="op">=</span><span class="va">ortholog.dict</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># For example, if you want to explore LR interactions</span></span>
<span><span class="co"># you can proceed as showed before for a human dataset.</span></span>
<span></span>
<span><span class="co"># bsrinf.redBP &lt;- reduceToBestPathway(bsrinf)  </span></span>
<span><span class="co"># bsrsig.redBP &lt;- getLRGeneSignatures(bsrinf.redBP, qval.thres=0.001)</span></span>
<span><span class="co"># scoresLR &lt;- scoreLRGeneSignatures(bsrdm,bsrsig.redBP,name.by.pathway=FALSE)</span></span>
<span><span class="co"># simpleHeatmap(scoresLR[1:20,], file="sdc_scoreLR.pdf", column.names=TRUE, width=9, height=5, pointsize=8)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatial-transcriptomics">Spatial Transcriptomics<a class="anchor" aria-label="anchor" href="#spatial-transcriptomics"></a>
</h2>
<p><br></p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>We would like to thank a few people who contributed to this work : - Laia Bassaganyas for insightful comments on the manuscript. - Guillaume Tosato for his help with the figures. - Gauthier Gadouas for testing the software on different platforms.</p>
<p><br></p>
<p>Thank you for reading this guide and for using <code>BulkSignalR</code>.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.0 (2022-04-22)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 18.04.6 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] doParallel_1.0.17 iterators_1.0.14  foreach_1.5.2     kableExtra_1.3.4 </span></span>
<span><span class="co">## [5] igraph_1.3.4      dplyr_1.0.9       BulkSignalR_0.0.9 BiocStyle_2.25.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] utf8_1.2.2                  tidyselect_1.1.2           </span></span>
<span><span class="co">##   [3] htmlwidgets_1.5.4           grid_4.2.0                 </span></span>
<span><span class="co">##   [5] BiocParallel_1.31.8         Rtsne_0.16                 </span></span>
<span><span class="co">##   [7] devtools_2.4.3              munsell_0.5.0              </span></span>
<span><span class="co">##   [9] ScaledMatrix_1.5.0          codetools_0.2-18           </span></span>
<span><span class="co">##  [11] ragg_1.2.2                  statmod_1.4.36             </span></span>
<span><span class="co">##  [13] scran_1.25.0                withr_2.5.0                </span></span>
<span><span class="co">##  [15] colorspace_2.0-3            Biobase_2.57.1             </span></span>
<span><span class="co">##  [17] highr_0.9                   knitr_1.39                 </span></span>
<span><span class="co">##  [19] ggalluvial_0.12.3           rstudioapi_0.13            </span></span>
<span><span class="co">##  [21] orthogene_1.2.0             stats4_4.2.0               </span></span>
<span><span class="co">##  [23] SingleCellExperiment_1.19.0 ggsignif_0.6.3             </span></span>
<span><span class="co">##  [25] labeling_0.4.2              MatrixGenerics_1.9.1       </span></span>
<span><span class="co">##  [27] GenomeInfoDbData_1.2.8      farver_2.1.1               </span></span>
<span><span class="co">##  [29] pheatmap_1.0.12             rprojroot_2.0.3            </span></span>
<span><span class="co">##  [31] vctrs_0.4.1                 treeio_1.20.1              </span></span>
<span><span class="co">##  [33] generics_0.1.3              xfun_0.31                  </span></span>
<span><span class="co">##  [35] R6_2.5.1                    GenomeInfoDb_1.33.3        </span></span>
<span><span class="co">##  [37] clue_0.3-61                 rsvd_1.0.5                 </span></span>
<span><span class="co">##  [39] locfit_1.5-9.6              bitops_1.0-7               </span></span>
<span><span class="co">##  [41] cachem_1.0.6                gridGraphics_0.5-1         </span></span>
<span><span class="co">##  [43] DelayedArray_0.23.0         assertthat_0.2.1           </span></span>
<span><span class="co">##  [45] scales_1.2.0                gtable_0.3.0               </span></span>
<span><span class="co">##  [47] beachmat_2.13.4             processx_3.7.0             </span></span>
<span><span class="co">##  [49] rlang_1.0.4                 systemfonts_1.0.4          </span></span>
<span><span class="co">##  [51] splines_4.2.0               GlobalOptions_0.1.2        </span></span>
<span><span class="co">##  [53] rstatix_0.7.0               lazyeval_0.2.2             </span></span>
<span><span class="co">##  [55] broom_1.0.0                 BiocManager_1.30.18        </span></span>
<span><span class="co">##  [57] yaml_2.3.5                  abind_1.4-5                </span></span>
<span><span class="co">##  [59] backports_1.4.1             tools_4.2.0                </span></span>
<span><span class="co">##  [61] usethis_2.1.6               bookdown_0.27              </span></span>
<span><span class="co">##  [63] ggplotify_0.1.0             ggplot2_3.3.6              </span></span>
<span><span class="co">##  [65] gplots_3.1.3                ellipsis_0.3.2             </span></span>
<span><span class="co">##  [67] jquerylib_0.1.4             RColorBrewer_1.1-3         </span></span>
<span><span class="co">##  [69] BiocGenerics_0.43.0         stabledist_0.7-1           </span></span>
<span><span class="co">##  [71] sessioninfo_1.2.2           Rcpp_1.0.9                 </span></span>
<span><span class="co">##  [73] sparseMatrixStats_1.9.0     zlibbioc_1.43.0            </span></span>
<span><span class="co">##  [75] purrr_0.3.4                 RCurl_1.98-1.7             </span></span>
<span><span class="co">##  [77] ps_1.7.1                    prettyunits_1.1.1          </span></span>
<span><span class="co">##  [79] ggpubr_0.4.0                GetoptLong_1.0.5           </span></span>
<span><span class="co">##  [81] S4Vectors_0.35.1            SummarizedExperiment_1.27.1</span></span>
<span><span class="co">##  [83] cluster_2.1.3               fs_1.5.2                   </span></span>
<span><span class="co">##  [85] magrittr_2.0.3              data.table_1.14.2          </span></span>
<span><span class="co">##  [87] RSpectra_0.16-1             circlize_0.4.15            </span></span>
<span><span class="co">##  [89] matrixStats_0.62.0          pkgload_1.3.0              </span></span>
<span><span class="co">##  [91] patchwork_1.1.1             evaluate_0.15              </span></span>
<span><span class="co">##  [93] IRanges_2.31.0              shape_1.4.6                </span></span>
<span><span class="co">##  [95] compiler_4.2.0              tibble_3.1.8               </span></span>
<span><span class="co">##  [97] KernSmooth_2.23-20          SingleCellSignalR_1.9.0    </span></span>
<span><span class="co">##  [99] crayon_1.5.1                htmltools_0.5.3            </span></span>
<span><span class="co">## [101] ggfun_0.0.6                 tidyr_1.2.0                </span></span>
<span><span class="co">## [103] aplot_0.1.6                 DBI_1.1.3                  </span></span>
<span><span class="co">## [105] gprofiler2_0.2.1            ComplexHeatmap_2.13.0      </span></span>
<span><span class="co">## [107] MASS_7.3-57                 babelgene_22.3             </span></span>
<span><span class="co">## [109] Matrix_1.4-1                car_3.1-0                  </span></span>
<span><span class="co">## [111] cli_3.3.0                   metapod_1.5.0              </span></span>
<span><span class="co">## [113] GenomicRanges_1.49.0        pkgconfig_2.0.3            </span></span>
<span><span class="co">## [115] pkgdown_2.0.5               scuttle_1.7.2              </span></span>
<span><span class="co">## [117] plotly_4.10.0               xml2_1.3.3                 </span></span>
<span><span class="co">## [119] ggtree_3.4.1                svglite_2.1.0              </span></span>
<span><span class="co">## [121] bslib_0.3.1                 dqrng_0.3.0                </span></span>
<span><span class="co">## [123] multtest_2.53.0             webshot_0.5.3              </span></span>
<span><span class="co">## [125] XVector_0.37.0              rvest_1.0.2                </span></span>
<span><span class="co">## [127] yulab.utils_0.0.5           stringr_1.4.0              </span></span>
<span><span class="co">## [129] SIMLR_1.23.0                callr_3.7.1                </span></span>
<span><span class="co">## [131] digest_0.6.29               RcppAnnoy_0.0.19           </span></span>
<span><span class="co">## [133] pracma_2.3.8                rmarkdown_2.14             </span></span>
<span><span class="co">## [135] tidytree_0.3.9              edgeR_3.39.1               </span></span>
<span><span class="co">## [137] DelayedMatrixStats_1.19.0   gtools_3.9.3               </span></span>
<span><span class="co">## [139] rjson_0.2.21                lifecycle_1.0.1            </span></span>
<span><span class="co">## [141] nlme_3.1-158                jsonlite_1.8.0             </span></span>
<span><span class="co">## [143] BiocNeighbors_1.15.1        carData_3.0-5              </span></span>
<span><span class="co">## [145] desc_1.4.1                  viridisLite_0.4.0          </span></span>
<span><span class="co">## [147] limma_3.53.3                fansi_1.0.3                </span></span>
<span><span class="co">## [149] pillar_1.8.0                lattice_0.20-45            </span></span>
<span><span class="co">## [151] homologene_1.4.68.19.3.27   survival_3.3-1             </span></span>
<span><span class="co">## [153] fastmap_1.1.0               httr_1.4.3                 </span></span>
<span><span class="co">## [155] pkgbuild_1.3.1              glue_1.6.2                 </span></span>
<span><span class="co">## [157] remotes_2.4.2               png_0.1-7                  </span></span>
<span><span class="co">## [159] bluster_1.7.0               stringi_1.7.8              </span></span>
<span><span class="co">## [161] sass_0.4.1                  textshaping_0.3.6          </span></span>
<span><span class="co">## [163] BiocSingular_1.13.0         caTools_1.18.2             </span></span>
<span><span class="co">## [165] memoise_2.0.1               irlba_2.3.5                </span></span>
<span><span class="co">## [167] ape_5.6-2</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jacques Colinge, Jean-Philippe Villemin.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
