<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ BulkSignalR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BulkSignalR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/UserGuide.html">User's Guide</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jcolinge/BulkSignalR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="BSRDataModel-class.html">BSRDataModel-class</a></code> </p>
        </td>
        <td><p>BulkSignalR Data Model Object</p></td>
      </tr><tr><td>
          <p><code><a href="BSRInference-class.html">BSRInference-class</a></code> </p>
        </td>
        <td><p>BulkSignalR Inference Object</p></td>
      </tr><tr><td>
          <p><code><a href="BSRSignature-class.html">BSRSignature-class</a></code> </p>
        </td>
        <td><p>BulkSignalR ligand-receptor signature Object</p></td>
      </tr><tr><td>
          <p><code><a href="BulkSignalR.html">BulkSignalR</a></code> </p>
        </td>
        <td><p>BulkSignalR: A package to infer ligand-receptor interactions from bulk transcriptomics or proteomics</p></td>
      </tr><tr><td>
          <p><code><a href="LRinter-BSRInference-method.html">LRinter(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>LRinter accessor</p></td>
      </tr><tr><td>
          <p><code><a href="LRinter-set-BSRInference-method.html">`LRinter&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>LRinter setter (internal use only)</p></td>
      </tr><tr><td>
          <p><code><a href="convertToHuman.html">convertToHuman()</a></code> </p>
        </td>
        <td><p>Transpose To Human Gene Names</p></td>
      </tr><tr><td>
          <p><code><a href="dot-buildPermutatedCountMatrix.html">.buildPermutatedCountMatrix()</a></code> </p>
        </td>
        <td><p>Internal function to generate a randomized expression matrix</p></td>
      </tr><tr><td>
          <p><code><a href="dot-buildPermutationIndices.html">.buildPermutationIndices()</a></code> </p>
        </td>
        <td><p>Internal function to generate expression matrix permutation indices</p></td>
      </tr><tr><td>
          <p><code><a href="dot-cdfAlphaStable.html">.cdfAlphaStable()</a></code> </p>
        </td>
        <td><p>Internal function to compute a censored alpha-) stable CDF</p></td>
      </tr><tr><td>
          <p><code><a href="dot-cdfGaussian.html">.cdfGaussian()</a></code> </p>
        </td>
        <td><p>Internal function to compute a censored Gaussian CDF</p></td>
      </tr><tr><td>
          <p><code><a href="dot-cdfMixedGaussian.html">.cdfMixedGaussian()</a></code> </p>
        </td>
        <td><p>Internal function to compute a censored mixed-Gaussian CDF</p></td>
      </tr><tr><td>
          <p><code><a href="dot-checkReceptorSignaling.html">.checkReceptorSignaling()</a></code> </p>
        </td>
        <td><p>Internal function to check receptor signaling downstream</p></td>
      </tr><tr><td>
          <p><code><a href="dot-cutExtremeValues.html">.cutExtremeValues()</a></code> </p>
        </td>
        <td><p>Internal function to cut extreme values from a matrix</p></td>
      </tr><tr><td>
          <p><code><a href="dot-downstreamSignaling.html">.downstreamSignaling()</a></code> </p>
        </td>
        <td><p>Internal function to check receptor signaling downstream</p></td>
      </tr><tr><td>
          <p><code><a href="dot-edgesLRIntracell.html">.edgesLRIntracell()</a></code> </p>
        </td>
        <td><p>Internal function to generate a ligand-receptor-downstream signaling network</p></td>
      </tr><tr><td>
          <p><code><a href="dot-geneNameConversion.html">.geneNameConversion()</a></code> </p>
        </td>
        <td><p>Convert gene symbol to another organism</p></td>
      </tr><tr><td>
          <p><code><a href="dot-getAlphaStableParam.html">.getAlphaStableParam()</a></code> </p>
        </td>
        <td><p>Internal function to fit a censored (alpha-) stable distribution</p></td>
      </tr><tr><td>
          <p><code><a href="dot-getCorrelatedLR.html">.getCorrelatedLR()</a></code> </p>
        </td>
        <td><p>Get correlated ligand-receptor pairs.</p></td>
      </tr><tr><td>
          <p><code><a href="dot-getEmpiricalNull.html">.getEmpiricalNull()</a></code> </p>
        </td>
        <td><p>Sampling of correlations downstream the receptors null distribution</p></td>
      </tr><tr><td>
          <p><code><a href="dot-getEmpiricalNullCorrLR.html">.getEmpiricalNullCorrLR()</a></code> </p>
        </td>
        <td><p>Sampling of ligand-receptor correlation null distribution</p></td>
      </tr><tr><td>
          <p><code><a href="dot-getGaussianParam.html">.getGaussianParam()</a></code> </p>
        </td>
        <td><p>Internal function to fit a censored Gaussian distribution</p></td>
      </tr><tr><td>
          <p><code><a href="dot-getMixedGaussianParam.html">.getMixedGaussianParam()</a></code> </p>
        </td>
        <td><p>Internal function to fit a censored mixed-Gaussian distribution</p></td>
      </tr><tr><td>
          <p><code><a href="dot-pValuesLR.html">.pValuesLR()</a></code> </p>
        </td>
        <td><p>Internal function to assign P-values to LR interactions</p></td>
      </tr><tr><td>
          <p><code><a href="dot-shufflePermutationIndices.html">.shufflePermutationIndices()</a></code> </p>
        </td>
        <td><p>Internal function to shuffle permutation indices</p></td>
      </tr><tr><td>
          <p><code><a href="dualHeatmap.html">dualHeatmap()</a></code> </p>
        </td>
        <td><p>Heatmap function for LR scores and additional data</p></td>
      </tr><tr><td>
          <p><code><a href="findOrthoGenes.html">findOrthoGenes()</a></code> </p>
        </td>
        <td><p>Orthologs Gene Names</p></td>
      </tr><tr><td>
          <p><code><a href="getLRGeneSignatures-BSRInference-method.html">getLRGeneSignatures(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Extract gene signatures of LR pair activity</p></td>
      </tr><tr><td>
          <p><code><a href="getLRIntracellNetwork.html">getLRIntracellNetwork()</a></code> </p>
        </td>
        <td><p>Generate a ligand-receptor-downstream signaling network</p></td>
      </tr><tr><td>
          <p><code><a href="getLRNetwork.html">getLRNetwork()</a></code> </p>
        </td>
        <td><p>Generate a ligand-receptor network</p></td>
      </tr><tr><td>
          <p><code><a href="getPathwayStats-BSRInference-method.html">getPathwayStats(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Basic statistics about hit pathways</p></td>
      </tr><tr><td>
          <p><code><a href="immune.signatures.html">immune.signatures</a></code> </p>
        </td>
        <td><p>Immune cell gene signatures</p></td>
      </tr><tr><td>
          <p><code><a href="infParam-BSRInference-method.html">infParam(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Inference parameters accessor</p></td>
      </tr><tr><td>
          <p><code><a href="infParam-set-BSRInference-method.html">`infParam&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Inference parameters setter (internal use only)</p></td>
      </tr><tr><td>
          <p><code><a href="initialInference-BSRDataModel-method.html">initialInference(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>Inference of ligand-receptor interactions</p></td>
      </tr><tr><td>
          <p><code><a href="initialOrganism-BSRDataModel-method.html">initialOrganism(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>organism accessor</p></td>
      </tr><tr><td>
          <p><code><a href="initialOrthologs-BSRDataModel-method.html">initialOrthologs(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>Model parameter accessor</p></td>
      </tr><tr><td>
          <p><code><a href="learnParameters-BSRDataModel-method.html">learnParameters(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>Training of BulkSignalR model parameters</p></td>
      </tr><tr><td>
          <p><code><a href="ligands-BSRInference-method.html">ligands(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>ligands accessor</p></td>
      </tr><tr><td>
          <p><code><a href="ligands-BSRSignature-method.html">ligands(<i>&lt;BSRSignature&gt;</i>)</a></code> </p>
        </td>
        <td><p>ligands accessor</p></td>
      </tr><tr><td>
          <p><code><a href="ligands-set-BSRInference-method.html">`ligands&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>ligands setter (internal use only)</p></td>
      </tr><tr><td>
          <p><code><a href="logTransformed-BSRDataModel-method.html">logTransformed(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>log.transformed accessor</p></td>
      </tr><tr><td>
          <p><code><a href="ncounts-BSRDataModel-method.html">ncounts(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>Normalized count matrix accessor</p></td>
      </tr><tr><td>
          <p><code><a href="ncounts-set-BSRDataModel-method.html">`ncounts&lt;-`(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>Normalized count matrix setter (internal use only)</p></td>
      </tr><tr><td>
          <p><code><a href="normalization-BSRDataModel-method.html">normalization(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>Normalization accessor</p></td>
      </tr><tr><td>
          <p><code><a href="param-BSRDataModel-method.html">param(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>Model parameter accessor</p></td>
      </tr><tr><td>
          <p><code><a href="pathways-BSRSignature-method.html">pathways(<i>&lt;BSRSignature&gt;</i>)</a></code> </p>
        </td>
        <td><p>pathways accessor</p></td>
      </tr><tr><td>
          <p><code><a href="prepareDataset.html">prepareDataset()</a></code> </p>
        </td>
        <td><p>Prepare a BSRDataModel object from expression data</p></td>
      </tr><tr><td>
          <p><code><a href="receptors-BSRInference-method.html">receptors(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>receptors accessor</p></td>
      </tr><tr><td>
          <p><code><a href="receptors-BSRSignature-method.html">receptors(<i>&lt;BSRSignature&gt;</i>)</a></code> </p>
        </td>
        <td><p>receptors accessor</p></td>
      </tr><tr><td>
          <p><code><a href="receptors-set-BSRInference-method.html">`receptors&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>receptors setter (internal use only)</p></td>
      </tr><tr><td>
          <p><code><a href="reduceToBestPathway-BSRInference-method.html">reduceToBestPathway(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Keep one pathway per ligand-receptor pair</p></td>
      </tr><tr><td>
          <p><code><a href="reduceToLigand-BSRInference-method.html">reduceToLigand(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Aggregate the receptors of a same ligand</p></td>
      </tr><tr><td>
          <p><code><a href="reduceToPathway-BSRInference-method.html">reduceToPathway(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Aggregate ligands and receptors at the pathway level</p></td>
      </tr><tr><td>
          <p><code><a href="reduceToReceptor-BSRInference-method.html">reduceToReceptor(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Aggregate the ligands of a same receptor</p></td>
      </tr><tr><td>
          <p><code><a href="rescoreInference-BSRInference-method.html">rescoreInference(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Inference re-scoring</p></td>
      </tr><tr><td>
          <p><code><a href="resetToInitialOrganism-BSRInference-method.html">resetToInitialOrganism(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p># Reset gene names to initial organism providen in first instance</p></td>
      </tr><tr><td>
          <p><code><a href="scoreLRGeneSignatures-BSRDataModel-method.html">scoreLRGeneSignatures(<i>&lt;BSRDataModel&gt;</i>)</a></code> </p>
        </td>
        <td><p>Score ligand-receptor gene signatures</p></td>
      </tr><tr><td>
          <p><code><a href="scoreSignatures.html">scoreSignatures()</a></code> </p>
        </td>
        <td><p>Generic gene signature scoring</p></td>
      </tr><tr><td>
          <p><code><a href="sdc.html">sdc</a></code> </p>
        </td>
        <td><p>Salivary duct carcinoma transcriptomes</p></td>
      </tr><tr><td>
          <p><code><a href="simpleHeatmap.html">simpleHeatmap()</a></code> </p>
        </td>
        <td><p>Heatmap function for LR scores</p></td>
      </tr><tr><td>
          <p><code><a href="tGenes-BSRInference-method.html">tGenes(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Target genes accessor</p></td>
      </tr><tr><td>
          <p><code><a href="tGenes-BSRSignature-method.html">tGenes(<i>&lt;BSRSignature&gt;</i>)</a></code> </p>
        </td>
        <td><p>Target genes accessor</p></td>
      </tr><tr><td>
          <p><code><a href="tGenes-set-BSRInference-method.html">`tGenes&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Target genes setter (internal use only)</p></td>
      </tr><tr><td>
          <p><code><a href="tgCorr-BSRInference-method.html">tgCorr(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Target gene correlations accessor</p></td>
      </tr><tr><td>
          <p><code><a href="tgCorr-set-BSRInference-method.html">`tgCorr&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> </p>
        </td>
        <td><p>Target gene correlations setter (internal use only)</p></td>
      </tr><tr><td>
          <p><code><a href="tme.signatures.html">tme.signatures</a></code> </p>
        </td>
        <td><p>Tumor microenvironment gene signatures</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacques Colinge, Jean-Philippe Villemin.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

