<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ BulkSignalR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">BulkSignalR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.9</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/BulkSignalR.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/jcolinge/BulkSignalR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="BSRDataModel-class.html">BSRDataModel-class</a></code> 
        </dt>
        <dd>BulkSignalR Data Model Object</dd>
      </dl><dl><dt>
          
          <code><a href="BSRInference-class.html">BSRInference-class</a></code> 
        </dt>
        <dd>BulkSignalR Inference Object</dd>
      </dl><dl><dt>
          
          <code><a href="BSRSignature-class.html">BSRSignature-class</a></code> 
        </dt>
        <dd>BulkSignalR ligand-receptor signature Object</dd>
      </dl><dl><dt>
          
          <code><a href="BulkSignalR.html">BulkSignalR</a></code> 
        </dt>
        <dd>BulkSignalR: A package to infer ligand-receptor interactions from bulk transcriptomics or proteomics</dd>
      </dl><dl><dt>
          
          <code><a href="LRinter-BSRInference-method.html">LRinter(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>LRinter accessor</dd>
      </dl><dl><dt>
          
          <code><a href="LRinter-set-BSRInference-method.html">`LRinter&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>LRinter setter (internal use only)</dd>
      </dl><dl><dt>
          
          <code><a href="alluvial.plot.html">alluvial.plot()</a></code> 
        </dt>
        <dd>Alluvial plot</dd>
      </dl><dl><dt>
          
          <code><a href="bodyMap.mouse.html">bodyMap.mouse</a></code> 
        </dt>
        <dd>Mouse transcriptomes across tissues</dd>
      </dl><dl><dt>
          
          <code><a href="bubblePlot.pathways.LR.html">bubblePlot.pathways.LR()</a></code> 
        </dt>
        <dd>Bubble Plot to explore LR &amp; Pathways</dd>
      </dl><dl><dt>
          
          <code><a href="chord.diagram.LR.html">chord.diagram.LR()</a></code> 
        </dt>
        <dd>Chord Diagram of LR interactions with correlations</dd>
      </dl><dl><dt>
          
          <code><a href="convertToHuman.html">convertToHuman()</a></code> 
        </dt>
        <dd>Transpose to Human Gene Names</dd>
      </dl><dl><dt>
          
          <code><a href="dot-buildPermutatedCountMatrix.html">.buildPermutatedCountMatrix()</a></code> 
        </dt>
        <dd>Internal function to generate a randomized expression matrix</dd>
      </dl><dl><dt>
          
          <code><a href="dot-buildPermutationIndices.html">.buildPermutationIndices()</a></code> 
        </dt>
        <dd>Internal function to generate expression matrix permutation indices</dd>
      </dl><dl><dt>
          
          <code><a href="dot-cdfAlphaStable.html">.cdfAlphaStable()</a></code> 
        </dt>
        <dd>Internal function to compute a censored alpha-) stable CDF</dd>
      </dl><dl><dt>
          
          <code><a href="dot-cdfGaussian.html">.cdfGaussian()</a></code> 
        </dt>
        <dd>Internal function to compute a censored Gaussian CDF</dd>
      </dl><dl><dt>
          
          <code><a href="dot-cdfMixedGaussian.html">.cdfMixedGaussian()</a></code> 
        </dt>
        <dd>Internal function to compute a censored mixed-Gaussian CDF</dd>
      </dl><dl><dt>
          
          <code><a href="dot-checkReceptorSignaling.html">.checkReceptorSignaling()</a></code> 
        </dt>
        <dd>Internal function to check receptor signaling downstream</dd>
      </dl><dl><dt>
          
          <code><a href="dot-customheatmap.html">.customheatmap()</a></code> 
        </dt>
        <dd>Heatmap function for gene expression of signature</dd>
      </dl><dl><dt>
          
          <code><a href="dot-cutExtremeValues.html">.cutExtremeValues()</a></code> 
        </dt>
        <dd>Internal function to cut extreme values from a matrix</dd>
      </dl><dl><dt>
          
          <code><a href="dot-downstreamSignaling.html">.downstreamSignaling()</a></code> 
        </dt>
        <dd>Internal function to check receptor signaling downstream</dd>
      </dl><dl><dt>
          
          <code><a href="dot-edgesLRIntracell.html">.edgesLRIntracell()</a></code> 
        </dt>
        <dd>Internal function to generate a ligand-receptor-downstream signaling network</dd>
      </dl><dl><dt>
          
          <code><a href="dot-geneNameConversion.html">.geneNameConversion()</a></code> 
        </dt>
        <dd>Convert gene symbol to another organism</dd>
      </dl><dl><dt>
          
          <code><a href="dot-getAlphaStableParam.html">.getAlphaStableParam()</a></code> 
        </dt>
        <dd>Internal function to fit a censored (alpha-) stable distribution</dd>
      </dl><dl><dt>
          
          <code><a href="dot-getCorrelatedLR.html">.getCorrelatedLR()</a></code> 
        </dt>
        <dd>Get correlated ligand-receptor pairs.</dd>
      </dl><dl><dt>
          
          <code><a href="dot-getEmpiricalNull.html">.getEmpiricalNull()</a></code> 
        </dt>
        <dd>Sampling of correlations downstream the receptors null distribution</dd>
      </dl><dl><dt>
          
          <code><a href="dot-getEmpiricalNullCorrLR.html">.getEmpiricalNullCorrLR()</a></code> 
        </dt>
        <dd>Sampling of ligand-receptor correlation null distribution</dd>
      </dl><dl><dt>
          
          <code><a href="dot-getGaussianParam.html">.getGaussianParam()</a></code> 
        </dt>
        <dd>Internal function to fit a censored Gaussian distribution</dd>
      </dl><dl><dt>
          
          <code><a href="dot-getMixedGaussianParam.html">.getMixedGaussianParam()</a></code> 
        </dt>
        <dd>Internal function to fit a censored mixed-Gaussian distribution</dd>
      </dl><dl><dt>
          
          <code><a href="dot-pValuesLR.html">.pValuesLR()</a></code> 
        </dt>
        <dd>Internal function to assign P-values to LR interactions</dd>
      </dl><dl><dt>
          
          <code><a href="dot-shufflePermutationIndices.html">.shufflePermutationIndices()</a></code> 
        </dt>
        <dd>Internal function to shuffle permutation indices</dd>
      </dl><dl><dt>
          
          <code><a href="dualHeatmap.html">dualHeatmap()</a></code> 
        </dt>
        <dd>Heatmap function for LR scores and additional data</dd>
      </dl><dl><dt>
          
          <code><a href="findOrthoGenes.html">findOrthoGenes()</a></code> 
        </dt>
        <dd>Orthologs Gene Names</dd>
      </dl><dl><dt>
          
          <code><a href="getLRGeneSignatures-BSRInference-method.html">getLRGeneSignatures(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Extract gene signatures of LR pair activity</dd>
      </dl><dl><dt>
          
          <code><a href="getLRIntracellNetwork.html">getLRIntracellNetwork()</a></code> 
        </dt>
        <dd>Generate a ligand-receptor-downstream signaling network</dd>
      </dl><dl><dt>
          
          <code><a href="getLRNetwork.html">getLRNetwork()</a></code> 
        </dt>
        <dd>Generate a ligand-receptor network</dd>
      </dl><dl><dt>
          
          <code><a href="getPathwayStats-BSRInference-method.html">getPathwayStats(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Basic statistics about hit pathways</dd>
      </dl><dl><dt>
          
          <code><a href="immune.signatures.html">immune.signatures</a></code> 
        </dt>
        <dd>Immune cell gene signatures</dd>
      </dl><dl><dt>
          
          <code><a href="infParam-BSRInference-method.html">infParam(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Inference parameters accessor</dd>
      </dl><dl><dt>
          
          <code><a href="infParam-set-BSRInference-method.html">`infParam&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Inference parameters setter (internal use only)</dd>
      </dl><dl><dt>
          
          <code><a href="initialInference-BSRDataModel-method.html">initialInference(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>Inference of ligand-receptor interactions</dd>
      </dl><dl><dt>
          
          <code><a href="initialOrganism-BSRDataModel-method.html">initialOrganism(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>organism accessor</dd>
      </dl><dl><dt>
          
          <code><a href="initialOrthologs-BSRDataModel-method.html">initialOrthologs(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>Model parameter accessor</dd>
      </dl><dl><dt>
          
          <code><a href="learnParameters-BSRDataModel-method.html">learnParameters(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>Training of BulkSignalR model parameters</dd>
      </dl><dl><dt>
          
          <code><a href="ligands-BSRInference-method.html">ligands(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>ligands accessor</dd>
      </dl><dl><dt>
          
          <code><a href="ligands-BSRSignature-method.html">ligands(<i>&lt;BSRSignature&gt;</i>)</a></code> 
        </dt>
        <dd>ligands accessor</dd>
      </dl><dl><dt>
          
          <code><a href="ligands-set-BSRInference-method.html">`ligands&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>ligands setter (internal use only)</dd>
      </dl><dl><dt>
          
          <code><a href="logTransformed-BSRDataModel-method.html">logTransformed(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>log.transformed accessor</dd>
      </dl><dl><dt>
          
          <code><a href="ncounts-BSRDataModel-method.html">ncounts(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>Normalized count matrix accessor</dd>
      </dl><dl><dt>
          
          <code><a href="ncounts-set-BSRDataModel-method.html">`ncounts&lt;-`(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>Normalized count matrix setter (internal use only)</dd>
      </dl><dl><dt>
          
          <code><a href="normalization-BSRDataModel-method.html">normalization(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>Normalization accessor</dd>
      </dl><dl><dt>
          
          <code><a href="param-BSRDataModel-method.html">param(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>Model parameter accessor</dd>
      </dl><dl><dt>
          
          <code><a href="pathways-BSRSignature-method.html">pathways(<i>&lt;BSRSignature&gt;</i>)</a></code> 
        </dt>
        <dd>pathways accessor</dd>
      </dl><dl><dt>
          
          <code><a href="prepareDataset.html">prepareDataset()</a></code> 
        </dt>
        <dd>Prepare a BSRDataModel object from expression data</dd>
      </dl><dl><dt>
          
          <code><a href="receptors-BSRInference-method.html">receptors(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>receptors accessor</dd>
      </dl><dl><dt>
          
          <code><a href="receptors-BSRSignature-method.html">receptors(<i>&lt;BSRSignature&gt;</i>)</a></code> 
        </dt>
        <dd>receptors accessor</dd>
      </dl><dl><dt>
          
          <code><a href="receptors-set-BSRInference-method.html">`receptors&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>receptors setter (internal use only)</dd>
      </dl><dl><dt>
          
          <code><a href="reduceToBestPathway-BSRInference-method.html">reduceToBestPathway(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Keep one pathway per ligand-receptor pair</dd>
      </dl><dl><dt>
          
          <code><a href="reduceToLigand-BSRInference-method.html">reduceToLigand(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Aggregate the receptors of a same ligand</dd>
      </dl><dl><dt>
          
          <code><a href="reduceToPathway-BSRInference-method.html">reduceToPathway(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Aggregate ligands and receptors at the pathway level</dd>
      </dl><dl><dt>
          
          <code><a href="reduceToReceptor-BSRInference-method.html">reduceToReceptor(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Aggregate the ligands of a same receptor</dd>
      </dl><dl><dt>
          
          <code><a href="rescoreInference-BSRInference-method.html">rescoreInference(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Inference re-scoring</dd>
      </dl><dl><dt>
          
          <code><a href="resetToInitialOrganism-BSRInference-method.html">resetToInitialOrganism(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd># Reset gene names to initial organism providen in first instance</dd>
      </dl><dl><dt>
          
          <code><a href="scoreLRGeneSignatures-BSRDataModel-method.html">scoreLRGeneSignatures(<i>&lt;BSRDataModel&gt;</i>)</a></code> 
        </dt>
        <dd>Score ligand-receptor gene signatures</dd>
      </dl><dl><dt>
          
          <code><a href="scoreSignatures.html">scoreSignatures()</a></code> 
        </dt>
        <dd>Generic gene signature scoring</dd>
      </dl><dl><dt>
          
          <code><a href="sdc.html">sdc</a></code> 
        </dt>
        <dd>Salivary duct carcinoma transcriptomes</dd>
      </dl><dl><dt>
          
          <code><a href="signatureHeatmaps.html">signatureHeatmaps()</a></code> 
        </dt>
        <dd>Heatmap function for gene expression of signature</dd>
      </dl><dl><dt>
          
          <code><a href="simpleHeatmap.html">simpleHeatmap()</a></code> 
        </dt>
        <dd>Heatmap function for LR scores</dd>
      </dl><dl><dt>
          
          <code><a href="tGenes-BSRInference-method.html">tGenes(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Target genes accessor</dd>
      </dl><dl><dt>
          
          <code><a href="tGenes-BSRSignature-method.html">tGenes(<i>&lt;BSRSignature&gt;</i>)</a></code> 
        </dt>
        <dd>Target genes accessor</dd>
      </dl><dl><dt>
          
          <code><a href="tGenes-set-BSRInference-method.html">`tGenes&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Target genes setter (internal use only)</dd>
      </dl><dl><dt>
          
          <code><a href="tgCorr-BSRInference-method.html">tgCorr(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Target gene correlations accessor</dd>
      </dl><dl><dt>
          
          <code><a href="tgCorr-BSRSignature-method.html">tgCorr(<i>&lt;BSRSignature&gt;</i>)</a></code> 
        </dt>
        <dd>Target genes accessor</dd>
      </dl><dl><dt>
          
          <code><a href="tgCorr-set-BSRInference-method.html">`tgCorr&lt;-`(<i>&lt;BSRInference&gt;</i>)</a></code> 
        </dt>
        <dd>Target gene correlations setter (internal use only)</dd>
      </dl><dl><dt>
          
          <code><a href="tme.signatures.html">tme.signatures</a></code> 
        </dt>
        <dd>Tumor microenvironment gene signatures</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jacques Colinge, Jean-Philippe Villemin.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

    </footer></div>

  

  

  </body></html>

