<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prepare a BSRDataModel object from expression data — prepareDataset • BulkSignalR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare a BSRDataModel object from expression data — prepareDataset"><meta property="og:description" content="Take a matrix or data frame containing RNA sequencing,
microarray, or expression proteomics data and returns a BSRDataModel
object ready for subsequent training. Normally, BSRDataModel objects
are not instantiated directly, but through this function."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BulkSignalR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/UserGuide.html">User's Guide</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jcolinge/BulkSignalR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prepare a BSRDataModel object from expression data</h1>
    <small class="dont-index">Source: <a href="https://github.com/jcolinge/BulkSignalR/blob/HEAD/R/dataPrepare.R" class="external-link"><code>R/dataPrepare.R</code></a></small>
    <div class="hidden name"><code>prepareDataset.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Take a matrix or data frame containing RNA sequencing,
microarray, or expression proteomics data and returns a BSRDataModel
object ready for subsequent training. Normally, BSRDataModel objects
are not instantiated directly, but through this function.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">prepareDataset</span><span class="op">(</span>
  <span class="va">counts</span>,
  normalize <span class="op">=</span> <span class="cn">TRUE</span>,
  symbol.col <span class="op">=</span> <span class="cn">NULL</span>,
  min.count <span class="op">=</span> <span class="fl">10</span>,
  prop <span class="op">=</span> <span class="fl">0.1</span>,
  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UQ"</span>, <span class="st">"TC"</span><span class="op">)</span>,
  log.transformed <span class="op">=</span> <span class="cn">FALSE</span>,
  min.LR.found <span class="op">=</span> <span class="fl">80</span>,
  species <span class="op">=</span> <span class="st">"hsapiens"</span>,
  conversion.dict <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Gene.name <span class="op">=</span> <span class="st">"A"</span>, row.names <span class="op">=</span> <span class="st">"B"</span><span class="op">)</span>,
  UQ.pc <span class="op">=</span> <span class="fl">0.75</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>counts</dt>
<dd><p>A table or matrix of read counts.</p></dd>
<dt>normalize</dt>
<dd><p>A logical indicating whether <code>counts</code> should be
normalized according to <code>method</code> or if it was normalized beforehand.</p></dd>
<dt>symbol.col</dt>
<dd><p>The index of the column containing the gene symbols in case
those are not the row names of <code>counts</code> already.</p></dd>
<dt>min.count</dt>
<dd><p>The minimum read count of a gene to be considered expressed
in a sample.</p></dd>
<dt>prop</dt>
<dd><p>The minimum proportion of samples where a gene must be
expressed higher than <code>min.count</code> to keep that gene.</p></dd>
<dt>method</dt>
<dd><p>The normalization method ('UQ' for upper quartile or 'TC'
for total count).</p></dd>
<dt>log.transformed</dt>
<dd><p>A logical indicating whether expression data were
already log2-transformed, e.g., some microarray data.</p></dd>
<dt>min.LR.found</dt>
<dd><p>The minimum number of ligands or receptors found in
<code>count</code> row names after eliminating the rows containing too many
zeros according to <code>min.count</code> and <code>prop</code>.</p></dd>
<dt>UQ.pc</dt>
<dd><p>Percentile for upper-quartile normalization, number
between 0 and 1 (in case the default 0.75 - hence the name - is not
appropriate).</p></dd>
<dt>organism</dt>
<dd><p>Organism</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A BSRModelData object with empty model parameters.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The <code>counts</code> matrix or table should be provided with expression
  levels of protein coding genes in each samples (column) and
  <code>rownames(counts)</code> set to HUGO offcial gene symbols. For commodity, it
  is also possible to provide <code>counts</code> with the
  gene symbols stored in one of its columns. This column must be specified
  with <code>symbol.col</code>. In such a case, <code>prepareDataset</code> will extract
  this column and use it to set the row names. Because row names must be
  unique, <code>prepareDataset</code> will eliminate rows with duplicated gene
  symbols by keeping the rows with maximum average expression. Gene symbol
  duplication may occur in protein coding genes after genome alignment
  due to errors in genome feature annotation files (GTF/GFF), where a handful
  of deprecated gene annotations might remain, or
  some genes are not given their fully specific symbols. If your read count
  extraction pipeline does not take care of this phenomenon, the maximum mean
  expression selection strategy implemented here should solve this difficulty
  for the sake of inferring ligand-receptor interactions.</p>
<p>If <code>normalize</code> is <code>TRUE</code> then normalization is performed
  according to <code>method</code>. If those two simple methods are not satisfying,
  then it is possible to provide a pre-normalized matrix setting
  <code>normalize</code> to <code>FALSE</code>.</p>
<p>In case proteomic or microarray data are provided, <code>min.count</code> must be
  understood as its equivalent with respect to those data.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">'prepareDataset'</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "prepareDataset"</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sdc</span>,package<span class="op">=</span><span class="st">'BulkSignalR'</span><span class="op">)</span></span>
<span class="r-in"><span class="va">normal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^N"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">sdc</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">bsrdm</span> <span class="op">&lt;-</span> <span class="fu">prepareDataset</span><span class="op">(</span><span class="va">sdc</span><span class="op">[</span>,<span class="op">-</span><span class="va">normal</span><span class="op">]</span><span class="op">)</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacques Colinge, Jean-Philippe Villemin.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

